### PR TITLE
WIP - videoStream.stop()

### PR DESCRIPTION
**Pull request by orblivion**
From #22. Just want to put this out here for discussion before I spend too much time going down an undesired path.

### CODE REVIEW

File: static/js/webrtc.js [hunk 1]
```diff
@@ -137,16 +137,18 @@ var rtc = (function() {
...
     },
     activate: function() {
+      var p = Promise.resolve()
       $("#options-enablertc").prop("checked", true);
       if (isActive) return;
       self.show();
       if (isSupported) {
         padcookie.setPref("rtcEnabled", true);
-        self.getUserMedia();
+        p = self.getUserMedia();
```
Comment (orblivion) [inline static/js/webrtc.js:146]: So that I can `.then()` below.

File: static/js/webrtc.js [hunk 2]
```diff
@@ -182,8 +184,24 @@ var rtc = (function() {
...
+        if (videoTrack.enabled) {
+          videoTrack.enabled = false;
+          videoTrack.stop()
+          return false
+        } else {
+          var audioTrack = localStream.getAudioTracks()[0];
+          console.log(1, audioTrack)
+          var audioTrackEnabled = audioTrack.enabled;
+          self.deactivate()
+          self.activate()
```
Comment (orblivion) [inline static/js/webrtc.js:196]: I've been experimenting; this is the only way I can figure out so far, how to recover from `.stop()`. I can always try a stripped down version of them as well. Right now I'm experimenting with just caling `getUserMedia` and adding the video to the existing stream. The whole hierarchy of things is a bit confusing so I don't know if that's appropriate though.

File: static/js/webrtc.js [hunk 3]
```diff
@@ -182,8 +184,24 @@ var rtc = (function() {
...
+          .then(() => {
+            // new one, now that we've reactivated
+            audioTrack = localStream.getAudioTracks()[0];
```
Comment (orblivion) [inline static/js/webrtc.js:199]: Assuming I do re-activate, I have to return audio to the same state it was at before. Though, I don't have a lot of confidence that this works.

**Comment by JohnMcLear** [comment]
Lemme know when you are ready for me to review / test

**Comment by JohnMcLear** [comment]
Just got a conflict @orblivion - you aware of that?

**Comment by orblivion** [comment]
Thanks. I stepped away from this task for a second (as I mentioned in the error message issue.)

**Comment by orblivion** [comment]
(Make sure I squash before you merge this!)

**Comment by orblivion** [comment]
Putting off this task. See details from today in #22.

**Pull request closed without merging**